src/Chapter3.hs

```diff
@@ -971,6 +1079,30 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday = Monday
+             | Tuesday
+             | Wednesday
+             | Thursday
+             | Friday
+             | Saturday
+             | Sunday
+             deriving (Ord, Eq, Show, Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend Saturday = True
+isWeekend Sunday   = True
+isWeekend _        = False
+
+nextDay :: Weekday -> Weekday
+nextDay Sunday = Monday
+nextDay d = succ d
```

Comment:
Unfortunately, `succ` is not a safe function as it will fail on runtim e when given the max boundary element (in this case, Sunday). You need to catch this case separately.